### ISSUE TITLE
Reporter prints output only after all suites are finished

### ISSUE THREAD

**Issue by qtis** [webdriverio/webdriverio#5056]
**Describe your question with as much detail as possible**
After migration from webdriver 4 to 5 - reporter output is shown only after all suites are finished running. Before, it used to print "as it goes". How can I configure wdio to print reporter output to console as soon as single test is finished?

**Environment (please complete the following information):**
 - **WebdriverIO version:** 5.18.7
 - **Mode:** Wdio test runner (local)
 - **If WDIO Testrunner, running sync/async:** sync
 - **Node.js version:** 10.19.0
 - **NPM version:** 6.13.4
 - **Browser name and version:** Chrome 80
 - **Platform name and version:** Windows 10
 - **Additional wdio packages used (if applicable):** @wdio/spec reporter, wdio-teamcity-reporter

**Additional context**

I've looked into the code and found that CliInterface would store messages from reporters unless you run wdio in --watch mode. Does it mean there's a need for feature request?

**Comment by christian-bromann** [comment]
> Does it mean there's a need for feature request?

What kind of behavior are you expecting? What is the value of seeing the spec reporter report after the test finishes?

**Comment by qtis** [comment]
> > Does it mean there's a need for feature request?
> 
> What kind of behavior are you expecting? What is the value of seeing the spec reporter report after the test finishes?

In our particular scenario we are using wdio-teamcity-reporter and as test-suite is being run - you can instantly see on progress how many tests are already executed and if something got skipped/errored in some suite - you can instantly react without the need to wait for full execution to finish.

Another value would be that in case of process kill because of whatever reasons - you are left with nothing.

**Comment by christian-bromann** [comment]
> Another value would be that in case of process kill because of whatever reasons - you are left with nothing.

What do you mean by that?

I think generally it makes sense to me to print reports of a worker once it finishes. We could even make this configurable. Any PRs for this would be appreciated.

**Comment by qtis** [comment]
I mean that if your all test cases combined would run 20minutes and at 19th minute you would receive some sort of out of memory exception - results would be gone because they are printed only when all tests are finished.

Normally I'd expect that as soon single test case finishes (or let's say in cucumber's runner single feature that includes several scenarios) - result gets printed.

Not sure what you are calling worker. A runner that runs all the tests found (be it thousands) or a single test run.

Here's the test that checks it is printer after each message received, but only in watch mode:
https://github.com/webdriverio/webdriverio/blob/c4def4c0db58edb2109a8b0c810f01ab73b730aa/packages/wdio-cli/tests/interface.test.js#L109

Let's make it configurable so it is possible to print it after each message without watch mode too! :)

**Comment by christian-bromann** [comment]
> If worker for you is a single test run then we are talking about the same thing :)

We do talk about the same thing. A worker in WebdriverIO is currently the Node process that runs the framework and contains the report logic.

> Let's make it configurable so it is possible to print it after each message without watch mode too! :)

Sounds good to me 👍

**Comment by mgrybyk** [comment]
I think this is partly related to https://github.com/webdriverio/webdriverio/issues/4547

It's pretty easy change to print messages after each spec file or even straightaway, all the logic is in interface.js

**Comment by qtis** [comment]
Yes, that discussion is about the same thing - adding a way to configure how reporters are printing out results, after each test or in the end.

**Comment by adrian-brown** [comment]
Previously the teamcity reporter did show the tests during execution e.g. tests passed / failed in TC.  Now has to wait until the end to get a view of results, that was in v4 though.  Likely v5 its not as up to date as the other reporters.
